wordpress-mobile/WordPress-Android
# Removes Land In the Editor ExPlat experiment

## Description
This PR replaces the [Land on the editor ExPlat experiment ](https://github.com/wordpress-mobile/WordPress-Android/pull/15627)with a development feature flag. This is needed to avoid pinging the ExPlat server now that the experiment is finished

## To test:
Use the test cases of the [A/B test Land In the Editor](https://github.com/wordpress-mobile/WordPress-Android/pull/15627) PR verifying that:
* when the `LAND_ON_THE_EDITOR` flag is **disabled** the **control** flow applies
* when the `LAND_ON_THE_EDITOR` flag is **enabled** the **treatment** flow applies

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
Updated existing `MySiteViewModelTest` test

PR submission checklist:
